packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2849,4 +2844,36 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.inAbstractClass = oldInAbstractClass;
       }
     }
+
+    tsParseAbstractDeclaration(
+      node: any,
+    ): N.ClassDeclaration | N.TsInterfaceDeclaration {
+      if (this.match(tt._class)) {
+        node.abstract = true;
+        return this.parseClass(
+          (node: N.ClassDeclaration),
+          /* isStatement */ true,
+          /* optionalId */ false,
+        );
+      } else if (this.isContextual("interface")) {
+        // for invalid abstract interface
+
+        // To avoid
+        //   abstract interface
+        //   Foo {}
+        if (!this.hasPrecedingLineBreak(this.lookahead())) {
```

Comment:
Is there a way to avoid this `lookahead`?

Reply:
Maybe with a combination of `lineBreak.test` and `this.nextTokenStart`

Reply:
It seems to work fine! da8ce69